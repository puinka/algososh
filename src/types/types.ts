import { ElementStates } from "./element-states";

export type TLetter = {
  char: string;
  state: ElementStates;
};

export type TElement = {
  value: number;
  state: ElementStates;
};
