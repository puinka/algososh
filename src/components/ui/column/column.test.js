import renderer from "react-test-renderer";
import { ElementStates } from "../../../types/element-states";
import { Column } from "./column";

describe("should render Column component", () => {
  it("without props", () => {
    const column = renderer.create(<Column />).toJSON();
    expect(column).toMatchSnapshot();
  });

  it("as default state", () => {
    const column = renderer
      .create(<Column state={ElementStates.Default} />)
      .toJSON();
    expect(column).toMatchSnapshot();
  });

  it("as changing state", () => {
    const column = renderer
      .create(<Column state={ElementStates.Changing} />)
      .toJSON();
    expect(column).toMatchSnapshot();
  });

  it("as modified state", () => {
    const column = renderer
      .create(<Column state={ElementStates.Modified} />)
      .toJSON();
    expect(column).toMatchSnapshot();
  });
});
