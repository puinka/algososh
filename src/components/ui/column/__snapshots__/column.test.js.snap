// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Column component as changing state 1`] = `
<div
  className="content "
>
  <div
    className="column changing"
    style={
      Object {
        "height": 1,
      }
    }
  />
  <p
    className="text text_type_column text_color_input mt-3"
  />
</div>
`;

exports[`should render Column component as default state 1`] = `
<div
  className="content "
>
  <div
    className="column default"
    style={
      Object {
        "height": 1,
      }
    }
  />
  <p
    className="text text_type_column text_color_input mt-3"
  />
</div>
`;

exports[`should render Column component as modified state 1`] = `
<div
  className="content "
>
  <div
    className="column modified"
    style={
      Object {
        "height": 1,
      }
    }
  />
  <p
    className="text text_type_column text_color_input mt-3"
  />
</div>
`;

exports[`should render Column component without props 1`] = `
<div
  className="content "
>
  <div
    className="column default"
    style={
      Object {
        "height": 1,
      }
    }
  />
  <p
    className="text text_type_column text_color_input mt-3"
  />
</div>
`;
